src/allotropy/parsers/utils/values.py

```diff
@@ -46,7 +60,7 @@ def try_float_or_none(value: Optional[str]) -> Optional[float]:
 
 
 def try_float_or_nan(value: Optional[str]) -> JsonFloat:
-    float_value = try_float_or_none(value)
+    float_value = try_non_nan_float_or_none(value)
```

Comment:
@alejandro-salgado is this really necesary? i.e. do we get NaN values from any of the instrument files?

My understanding is that it is either a float or `None`

Reply:
this is a safe guard in case there is a value that is read as nan.

Reply:
I don't really think this is necessary `try_float_or_none` should be enough as in all the other cases. It is very unlikely we would end up with something that is going to be serialized as `NaN` by the parsers unless we explicitly pass `np.nan` or `float(nan)`

Reply:
The idea of this function to be used anywhere, if we do not check for nan value then the rest of the logic will not produce the correct result. If we read a float(nan) value with try_float_or_none the result will be float(nan) and since that is not None the result of the function will be float(nan) and it should be "NaN"